inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/pooling/pooling_kernel_gpu_b_fs_zyx_fsv16_imad.cpp

```diff
@@ -54,12 +54,25 @@ PoolingKernelBase::DispatchData PoolingKernelGPU_b_fs_zyx_fsv16_imad::SetDefault
     auto f = out.Feature().v;
     auto b = out.Batch().v;
 
-    dispatchData.gws[0] = x;
-    dispatchData.gws[1] = y * z;
-    // we got b_fs_yx_fsv16 format, we process 16 features per workitem
-    dispatchData.gws[2] = CeilDiv(f, FEATURE_SLICE_SIZE) * b;
-    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    if (IsGlobalPooling(params)) {
+        y = params.inputs[0].Y().v;
+        z = params.inputs[0].Z().v;
+        dispatchData.gws[0] = x;
+        dispatchData.gws[1] = std::min(static_cast<int>(y * z), 256);
```

Comment:
You should also align global work size to 16, otherwise it may provoke errors in some cases